### PR TITLE
added swipe matching card component.

### PR DESCRIPTION
**Pull request by Sarnet2020**

### CODE REVIEW

File: gui/app/(tabs)/index.tsx [hunk 1]
```diff
@@ -1,93 +1,296 @@
...
+      id: 1,
+      name: "Jane",
+      age: 25,
+      interests: ["Hiking", "Food", "Reading"],
+      imageUrl:
+        "https://images.unsplash.com/photo-1678875922894-7d3210b0787d?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80",
+    },
+    // add more users here...
+  ];
+  return DEMO_CONTENT;
```
Comment (elliottallmann) [inline gui/app/(tabs)/index.tsx]: Please use actual API endpoints for getting the data. You can seed the database manually.

File: gui/app/(tabs)/index.tsx [hunk 2]
```diff
@@ -1,93 +1,296 @@
...
+      <Text style={styles.swipeText}>
+        User Liked: {usersLiked.length}
+        User Disliked :{usersDisliked.length} 
+      </Text>
+    </SafeAreaView>
   );
-}
+};
+
+export default Swipe;
```
Comment (elliottallmann) [inline gui/app/(tabs)/index.tsx]: This should be its own component under the `/components` directory. Please see how the other pages are structured so far. Happy to sit down and pair on this if it makes it easier.

Comment (elliottallmann) [reply gui/app/(tabs)/index.tsx]: Arguably this file should be split into at least 3 different components.

File: gui/app/(tabs)/index.tsx (hunk 1 above)
Comment (Sarnet2020) [reply gui/app/(tabs)/index.tsx]: can you provide me the API URL I can fetch data from backend like we did registration form.

Comment (elliottallmann) [reply gui/app/(tabs)/index.tsx]: Make a GET request to "/api/user/interest/{email}", where {email} is the email address of the user you are getting. Alternatively, pass the ID.

Expect a response that looks like our other API responses: 
```
{ 
  'error' => 'Error message here for display'
}
```
or 
```
{
  'success' => [
    {
      'activity' => Activity,
      'attitude' => Attitude,
      'skillLevel' => SkillLevel
    },
    {
      'activity' => Activity,
      'attitude' => Attitude,
      'skillLevel' => SkillLevel
    },
    {
      'activity' => Activity,
      'attitude' => Attitude,
      'skillLevel' => SkillLevel
    }
   ...
  ]
}
```

So you would make two requests to get the information you need: 
* One to get the profile information, such as the name, age, location, and profile picture
* One to get their interests

You can make both requests at the same time and do a fork-join on them when they come back, or make the first request to build the card, wait for it to come back, and then update the component with the results of the second request via hooks. Up to you.

If you want, we can write a method to give you all of the data in one request.

File: gui/components/swipe/InterestsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { FlatList, StyleSheet, Text, View } from "react-native";
+
+const Item = ({ item }) => (
+  <View style={styles.item}>
+    <Text>{item.activity}</Text>
+    <Text>{item.attitude}</Text>
+    <Text>{item.skillLevel}</Text>
+  </View>
+);
```
Comment (elliottallmann) [inline gui/components/swipe/InterestsTable.tsx]: `Item` should be it's own component. Anything that could be used somewhere else or in a loop should be its own component in its own file.

File: gui/components/swipe/index.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from "react";
+import { SafeAreaView, StyleSheet, View, Text, Dimensions } from "react-native";
+import SwipeableCard from "../../components/swipe/SwipeableCard";
+import SwipeCount from "./swipeCounts";
+const SCREEN_WIDTH = Dimensions.get("window").width;
+
+const Swipe = ({ cards }) => {
```
Comment (elliottallmann) [inline gui/components/swipe/index.tsx]: Can you update the file name to something more specific than index? Is this a container that pulls in Swipeable cards? If this is a page definition, should it go under (tabs)?

**Pull request merged**